src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Unix.cs

```diff
@@ -77,6 +71,62 @@ private bool InitCultureDataCore()
             return true;
         }
 
+        /// <summary>
+        /// Implementation of culture name validation.
+        /// </summary>
+        /// <remarks>
+        /// This is a fast approximate implementation based on BCP47 spec. It covers only parts of
+        /// the spec; such that, when it returns false, the input is definitely in incorrect format.
+        /// However, it returns true for some characters, which are not allowed by the spec, It also
+        /// returns true for some inputs where spec specifies the lengths of components, but we are
+        /// not validating components individually to limit its complexity at O(n) for now.
+        ///
+        /// Rules of implementation:
+        /// * Allow only letters, digits, - and '_' or \0 (NULL is for backward compatibility).
+        /// * Allow input length of zero (for invariant culture) or otherwise greater than 1 and less than or equal 85.
+        /// * Disallow input that starts or ends with '-' or '_'
+        /// * Disallow input that has any combination of consecutive '-' or '_'.
+        /// * Disallow input that has multiple '_'.
+        /// </remarks>
+        private static bool IsValidCultureName(string subject, out int firstIndexOfUnderscore)
+        {
+            firstIndexOfUnderscore = -1;
+
+            if (subject.Length == 0) return true; // Invariant Culture
+            if (subject.Length == 1 || subject.Length > 85) return false;
```

Comment:
>85 [](start = 56, length = 2)

we already have this defined constant in CultureData class. please use the constant.

private const int LocaleNameMaxLength = 85;